### ISSUE TITLE
Back to Basics this year...

### ISSUE THREAD

**Issue by chipmonkey** [NaNoGenMo/2019#76]
I was maybe overly ambitious in years past... trying to formalize a bit more this time 'round...

My goal this year was to take some random text, character, and location generators and build a basic recursive quest engine.

Let's unpack that.  All I want for now is:

- locations - a small network of places, with the ability for characters to travel between them
- characters - a recurring finite set of characters who travel between the locations and interact when they're in the same place
- stretch goal - interact when they're passing one another!
- recursive - the idea is that we have a world... it has, say, cities, within cities are places (inns, bars, markets), within those may be rooms... or maybe our recursion is deep enough at two levels

Eh, maybe that's even too much.  I work a super lot and in eight days of November so far all I've done is written those bullets.  But let's see what's out there...

At the moment I'm thinking of tying TextWorld with GPT-2 to automate universe building and text generation.  TextWorld needs multi-character support, but it has a nice questing tree.  Borrow Borrow!

**Comment by chipmonkey** [comment]
Ok, some dev notes here: http://www.chiplynch.com/wiki/index.php/NaNoGenMo_2019_-_GPT-2_Edition

First draft here: http://www.chiplynch.com/wiki/index.php/NaNoGenMo2019_Attempt_1

Which is awful, but has a few gems:
"Never lose confidence. Sure, you've been auctioned off, money has gone bad, just keep on parking."

"And that is when it ends
For every time things melt out faster
I have always wantedlet mankind melt
Ain't everything burn to shreds when bursting apart,
And being lost apart through convenience achievements"

but it's not adhering to the storyline as much as I'd like.  Working on it.

**Comment by chipmonkey** [comment]
Success!  Mostly.  Kind of.

TLDR; 50k text is here: http://www.chiplynch.com/wiki/index.php/NaNoGenMo2019_Final_Text
Code is here: https://github.com/chipmonkey/ChipMonkeyNaNoGenMo2019

OK, was on call for work today so in between making sure things don't crash I polished this little dirt ball up as much as I think I can for the month, and I think it's getting shinier!

I'm cutting it off for now.   I have 50k words, reminiscent of the Brothers' Grimm, on whose prose the neural net was trained.  The previous plan still applies - generate a world with a story and a walkthrough in TextWorld then embellish the runthrough with gpt-2.

Instructions in the README.md

> When the old folks were gone, and the wood behind the mill was dry,
> the little girl sat down upon a branch and cried:
> 
>  ‘My mother, my dear boy,
>   Will you cry?’
> 
> And the mother answered: ‘Oh, yes, dear boy, you should.’

A few things... I lost Chapter formatting after the last version, because I switched from using GPT-2 directly to using gpt-2-simple and forgot to port that part over, but it shouldn't be hard.

And the story line still isn't real, but it's there, if you look really, REALLY hard for it.

**Comment by hugovk** [comment]
Well done!

I'll re-open this, we like to keep issues open so people can find entries easier.

**Comment by chipmonkey** [comment]
Thanks @hugovk!  Wasn't really following how to handle it, so I appreciate the tags and re-open.  'til next year!
